content/learn/book/ecs/_index.md

```diff
@@ -0,0 +1,140 @@
+In Rust, this corresponds to mutable access, as declared by the use of `Query<&mut MyComponent>` or `ResMut<MyResource>`.
+
+On the next page, we'll create a simple "game" using the ECS so you can see how this all fits together.
+
+## Rust tips and tricks for Bevy's ECS
+
+If you're new to Rust, there are a few standard tricks and idioms that are used by Bevy's ECS that are worth learning about.
+Feel free to come back to this section later as you learn more about Bevy!
+
+### Tuple structs
+
+When working with tuple structs like `struct Score(u64)` or `struct Position(u8, u8)`, `.0`, `.1` and so on are used to access fields, counting up from 0 in the order they were defined.
+
+`let foo.0 = bar` can seem opaque when you first read it, but in Rust it just means that you're assigning `bar` to the first field of the tuple struct `foo`.
+Using ordinary structs with named fields rather than tuple structs (even for simple components and resources!) can go a long way to improving the clarity of your code by providing more descriptive field names.
+
+### Smart pointers
+
+There are a few points to be aware of when working with "smart pointers" like components and resources:
+
+1. The value returned in a system when you ask for `Res<Score>` is not a plain `Score`, it is a `Res<Score>`. This is a resource that stores the score, similar to an ordinary `&` pointer, but with some special behavior (like change detection).
+2. Because `Res` and other smart pointers implement the `Deref` trait, Rust will attempt to automatically dereference it to match required types when it can.
+   1. This is commonly done when accessing fields and methods: calling `.title` on `Res<WindowDescriptor>` will grab the title field of `WindowDescriptor`, not the wrapping `Res<WindowDescriptor>` as the `Res` type has no field named `title`.
+3. When your types cannot be automatically dereferenced, you will need to manually dereference out of the outer smart pointer using `*`.
+   1. This is commonly needed when assigning a value to your data (rather than just one of its fields), or when using `match`.
+
+### Initializing structs with some default values
+
+Bevy often requires defining complex structs when creating entities with bundles or configuring the app with resources.
+In many cases, we want to use *some* default fields, but not all of them.
+As Rust does not have default arguments, we instead use [struct update syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) in combination with the [`default()`](https://doc.rust-lang.org/std/default/trait.Default.html) method from the `Default` trait to allow us to quickly set some values while leaving others unchanged.
```

Comment:
```suggestion
As Rust does not have default arguments, we instead use [struct update syntax (SUS)](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) in combination with the [`default()`](https://doc.rust-lang.org/std/default/trait.Default.html) method from the `Default` trait to allow us to quickly set some values while leaving others unchanged.
```

It was stronger than me lol. Please don't accept it 😝

Reply:
Since `default` is the *only* required method of the `Default` trait, this seems needlessly wordy. My suggestion would be something along the lines of:

> ".. we instead use struct update syntax in combination with [`Default::default()`](https://doc.rust-lang.org/std/default/trait.Default.html), which allows us to succinctly specify only the values we are interested in."

Reply:
> ".. we instead use struct update syntax in combination with Default::default(), which allows us to succinctly specify only the values we are interested in."

Stealing this bit :)